sysdig_client.go

```diff
@@ -0,0 +1,113 @@
+package sysdig_client
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/NeowayLabs/sysdig-client/client"
+)
+
+type Sysdigclient struct {
+	client client.Client
+}
+
+type Query struct {
+	Last    int      `json:"last"`
+	Metrics []Metric `json:"metrics"`
+	Filter  string   `json:"filter"`
+}
+
+type Metric struct {
+	Id           string      `json:"id"`
+	Aggregations Aggregation `json:"aggregations"`
+}
+
+type Aggregation struct {
+	Time  string `json:"time"`
+	Group string `json:"group"`
+}
+
+type Period struct {
+	Days    int `json:"days"`
+	Hours   int `json:"hours"`
+	Minutes int `json:"minutes"`
+}
+
+type ApiResult struct {
+	Data  []Data `json:"data"`
+	Start int    `json:"start"`
+	End   int    `json:"end"`
+}
+
+type Data struct {
+	Value []int `json:"d"`
+}
+
+func (s *Sysdigclient) GetSumMetric(metrics []Metric, filter string, period Period) (int, error) {
+
+	secondsPeriod, err := s.getPeriodInSeconds(period)
+
+	if err != nil {
+		return 0, fmt.Errorf("error on give a period in seconds: %s", err)
+	}
+
+	query := Query{
+		Last:    secondsPeriod,
+		Filter:  filter,
+		Metrics: metrics,
+	}
+
+	bodyValue, err := json.Marshal(query)
+
+	if err != nil {
+		return 0, fmt.Errorf("error on marshal query: %s", err)
+	}
+
+	response := s.client.DoRequest(
+		client.Request{
+			Method: "POST",
+			URI:    "/api/data",
+			Body:   bodyValue,
+		},
+	)
+
+	var result ApiResult
+	err = json.Unmarshal(response.Body, &result)
+
+	if err != nil {
+		return 0, fmt.Errorf("error on unmarshal response result: %s", err)
+	}
+
+	if len(result.Data) == 0 {
+		return 0, fmt.Errorf("api returned the blank data field")
+	}
+
+	return result.Data[0].Value[0], nil
+}
+
+func (s *Sysdigclient) getPeriodInSeconds(period Period) (int, error) {
```

Comment:
I use a struct so that it can be set 3 types of inputs: Days, Hours and Minutes. As it is a struct I require the Lib user to pass at least 1 of the fields and this also facilitates my verification later.